scipy/optimize/_minimize.py

```diff
@@ -924,14 +925,16 @@ def minimize_scalar(fun, bracket=None, bounds=None, args=(),
     if meth == '_custom':
         res = method(fun, args=args, bracket=bracket, bounds=bounds, **options)
     elif meth == 'brent':
-        res = _minimize_scalar_brent(fun, bracket, args, **options)
+        res = _recover_from_bracket_error(_minimize_scalar_brent,
```

Comment:
`bracket` now raises an error when it fails to find a bracket, but we want `minimize_scalar` to return gracefully with `success == False`. `_recover_from_bracket_error` tries to do the minimization to create an `OptimizeResult` with the correct solution, but if a `BracketError` occurs, it creates an `OptimizeResult` with the best information available from the bracket search (and `success == False`).